### PR TITLE
merge additions for identity cli api management

### PR DESCRIPTION
**Pull request by hckenmiller**
This change is [Reviewable](https://reviewable.io/reviews/healthcatalyst/fabric.identity/233)

### CODE REVIEW

File: Fabric.Identity.API/scripts/CatalystDosIdentity.psm1 [hunk 1]
```diff
@@ -528,27 +528,356 @@ function Test-IsClientRegistered {
...
+                # Reset api secret
+                $apiResponse = Reset-ApiPassword -identityUrl $url -apiName $($apiResourceObject.name) -accessToken $accessToken
+                return $apiResponse.apiSecret
+            }catch{
+                $error = "Unknown error attempting to post api"
+                $exception = $_.Exception
+                if ($null -ne $exception -and $null -ne $exception.Response) {
+                    $error = Get-ErrorFromResponse -response $exception.Response
+                }
+                throw (New-Object -TypeName "System.Net.WebException" "There was an error registering api $($apiResourceObject.name): $error. Halting installation.", $exception)
```
Comment (ctediz) [inline Fabric.Identity.API/scripts/CatalystDosIdentity.psm1]: This error should be updated to not say it's halting installation

File: Fabric.Identity.API/scripts/tests/CatalystDosIdentity.tests.ps1 [hunk 2]
```diff
@@ -451,12 +451,289 @@ Describe 'Test-IsClientRegistered Unit Tests' -tag 'Unit' {
...
+            Mock -ModuleName CatalystDosIdentity -CommandName Invoke-RestMethod { throw  New-Object -TypeName "System.Net.WebException" -ArgumentList "Exception" }
+
+            {Get-ApiRegistration -identityUrl $mockUrl -apiName "someNonExistentApi" -accessToken  "Bearer goodtoken"} | Should -Throw "Exception" -ExceptionType System.Net.WebException
+        }
+    }
+}
+
+Describe 'New-ApiRegistration Unit Tests' -tag "Unit" {
+    Context 'Valid Request' {
+        It 'Should Create and Return an api object when valid response' {
```
Comment (ctediz) [inline Fabric.Identity.API/scripts/tests/CatalystDosIdentity.tests.ps1]: looks like it's returning a secret and not an object

Comment (ryanorbaker) [reply Fabric.Identity.API/scripts/tests/CatalystDosIdentity.tests.ps1]: the secret is what is returned, not the whole object.

Comment (hckenmiller) [reply Fabric.Identity.API/scripts/tests/CatalystDosIdentity.tests.ps1]: yah. I will remove the word object and change to secret

File: Fabric.Identity.API/scripts/CatalystDosIdentity.psm1 (hunk 1 above)
Comment (hckenmiller) [reply Fabric.Identity.API/scripts/CatalystDosIdentity.psm1]: yah. That was part of the original script run in the installer.

**Pull request merged**